ksqldb-engine/src/main/java/io/confluent/ksql/services/DefaultConnectClientFactory.java

```diff
@@ -94,7 +111,7 @@ private void startBasicAuthFileWatcher(
       credentialsFileWatcher.start();
       log.info("Enabled automatic connector credentials reload for location: " + filePath);
     } catch (java.io.IOException e) {
-      log.error("Failed to enable automatic connector credentials reload.");
+      log.error("Failed to enable automatic connector credentials reload: " + e.getMessage());
```

Comment:
My 2c, feel free to ignore as I am probably splitting hairs here: it is more informative to pass an exception as the second argument to the server logs rather than concatting its message because you also get a stack trace. Having the stack trace in the logs makes debugging so much easier. Consider a (made up) example where I added a runtime exception somewhere in the along the call chain:

```java
  @Test
  public void test() {
    try {
      final KsqlEntity response = executeListVariables("LIST VARIABLES;").get();
    } catch (Exception e) {
      logger.error("logging with the second argument", e);
      logger.error("logging by concating the message: " + e.getMessage());
    }
  }
```

Test output:

```
[2021-12-13 19:23:45,956] ERROR logging with the second argument (io.confluent.ksql.rest.server.execution.ListVariablesExecutorTest:95)
java.lang.RuntimeException: :(
	at io.confluent.ksql.rest.server.execution.ListVariablesExecutor.execute(ListVariablesExecutor.java:42)
	at io.confluent.ksql.rest.server.execution.CustomExecutors.execute(CustomExecutors.java:124)
	at io.confluent.ksql.rest.server.execution.ListVariablesExecutorTest.executeListVariables(ListVariablesExecutorTest.java:55)
	at io.confluent.ksql.rest.server.execution.ListVariablesExecutorTest.test(ListVariablesExecutorTest.java:93)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.runners.DefaultInternalRunner$1$1.evaluate(DefaultInternalRunner.java:54)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.mockito.internal.runners.DefaultInternalRunner$1.run(DefaultInternalRunner.java:99)
	at org.mockito.internal.runners.DefaultInternalRunner.run(DefaultInternalRunner.java:105)
	at org.mockito.internal.runners.StrictRunner.run(StrictRunner.java:40)
	at org.mockito.junit.MockitoJUnitRunner.run(MockitoJUnitRunner.java:163)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
[2021-12-13 19:23:45,963] ERROR logging by concating the message: :( (io.confluent.ksql.rest.server.execution.ListVariablesExecutorTest:96)
```

Even though the first option is way more verbose and contains a bunch of entries we do not care about, the top N entries are pure gold - they show us where _exactly_ the exception has happened.

Reply:
Makes sense -- thanks for the tip! I will make this update in my follow-up (dependent) PR (https://github.com/confluentinc/ksql/pull/8357) rather than this PR in order to avoid waiting for another build before merging.